apps/web/src/screens/Request/Signature/index.tsx

```diff
@@ -76,8 +79,9 @@ export const RequestSignature = () => {
 	}, []);
 
 	useEffect(() => {
-		getMessageOrTransaction(requestId).then((result) => {
+		getDataFromSourceRequest(requestId).then((result) => {
 			setPayload(result as MessagePayload);
+			setSender(result?.sender);
```

Comment:
refactor all similar places in to a re-usable Effect, e.g `useSourceRequestData`